### PR TITLE
Localise log boundary methods to preserve log source information

### PR DESCRIPTION
**Pull request by chrisseaton**

### CODE REVIEW

File: src/main/java/org/truffleruby/debug/TruffleDebugNodes.java [hunk 1]
```diff
@@ -300,10 +300,15 @@ protected int getCacheLimit() {
...
                 VirtualFrame frame, Object value,
                 @Cached("create()") NameToJavaStringNode toJavaStringNode) {
-            Log.warning(toJavaStringNode.executeToJavaString(frame, value));
+            warning(toJavaStringNode.executeToJavaString(frame, value));
             return nil();
         }
 
+        @TruffleBoundary
+        public static void warning(String format, Object... args) {
+            Log.LOGGER.warning(String.format(format, args));
```
Comment (eregon) [inline src/main/java/org/truffleruby/debug/TruffleDebugNodes.java]: This case does not need the `String.format()`

File: src/main/java/org/truffleruby/core/TruffleSystemNodes.java [hunk 2]
```diff
@@ -182,7 +182,7 @@ public Object fullMemoryBarrier() {
         public Object log(DynamicObject level, DynamicObject message,
```
Comment (eregon) [inline src/main/java/org/truffleruby/core/TruffleSystemNodes.java]: This Specialization will fail if more than 3 different levels  are passed (already a problem before).
Probably not common as the level is normally a literal Symbol.

Comment (chrisseaton) [inline src/main/java/org/truffleruby/core/TruffleSystemNodes.java]: Fixing.

File: src/main/java/org/truffleruby/debug/TruffleDebugNodes.java (hunk 1 above)
Comment (chrisseaton) [inline src/main/java/org/truffleruby/debug/TruffleDebugNodes.java]: Fixing.

**Comment by chrisseaton** [comment]
@eregon can you review again

**Pull request merged**